packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -570,21 +581,25 @@ export class ConversationService extends TypedEventEmitter<Events> {
           throw new Error('Qualified conversation id is missing in the event');
         }
 
-        const mlsConversation = await this.apiClient.api.conversation.getConversation(conversationId);
-
-        if (!isMLSConversation(mlsConversation)) {
-          throw new Error('Conversation is not an MLS conversation');
-        }
-
-        await this.handleConversationEpochMismatch(mlsConversation, () =>
-          this.emit('MLSConversationRecovered', {conversationId}),
-        );
+        await this.recoverMLSConversationFromEpochMismatch(conversationId);
         return;
       }
       throw error;
     }
   }
 
+  private async recoverMLSConversationFromEpochMismatch(conversationId: QualifiedId) {
+    const mlsConversation = await this.apiClient.api.conversation.getConversation(conversationId);
+
+    if (!isMLSConversation(mlsConversation)) {
+      throw new Error('Conversation is not an MLS conversation');
+    }
+
+    return this.handleConversationEpochMismatch(mlsConversation, () =>
```

Comment:
should this still await also?

Reply:
Nope, simple return is enough. Since we're returning async method here, it will get awaited anyway when parent method is awaited.